pkg/local_object_storage/blobstor/fstree/head.go

```diff
@@ -38,30 +39,36 @@ func (t *FSTree) Head(addr oid.Address) (*objectSDK.Object, error) {
 		}
 		return nil, fmt.Errorf("read file %q: %w", p, err)
 	}
-	defer f.Close()
 
-	obj, err := t.extractHeaderOnly(addr.Object(), f)
+	obj, reader, err := t.extractHeaderAndStream(addr.Object(), f)
+	if reader != nil {
+		_ = reader.Close()
```

Comment:
... although I see you have a problem of proper `f.Close()` in error cases and this _is_ a solution.